tools/tertiary-analysis/scmap/scmap_index_cluster.xml

```diff
@@ -6,9 +6,13 @@
     <expand macro="requirements" />
     <command detect_errors="exit_code"><![CDATA[
         scmap-preprocess-sce.R --input-object "${input_single_cell_experiment}" --output-sce-object "${input_single_cell_experiment}.preprocessed" && scmap-index-cluster.R --input-object-file "${input_single_cell_experiment}" --cluster-col '$cluster_col' --output-object-file '$output_single_cell_experiment' --output-plot-file '$plot'
```

Comment:
Maybe this was not part of this PR, but the way that the --output-sce-object is handled here is not the correct one. Could you please re-write it like `--output-sce-object "${output_single_cell_experiment}"` with an adequate entry in the `<outputs>`. If writing things explicitly not to `<outputs>` variables, that should only be done to the working directory and be discovered or moved to elements in `<outputs>`.

Reply:
These are actually two scripts at work here. `scmap-preprocess-sce.R` does some data normalisation and prepares the object. Now, the output entry corresponds to what is produced by `scmap-index-cluster.R `. If we make proper output entries for both, there will be two independent tools which would be inconvenient for the user.

Reply:
The problem is that `"${input_single_cell_experiment}.preprocessed"` is not something that you will be able to use as input to any other tools, as it is not exposed as an output on Galaxt. If this file doesn't need to be left as an output, then you just write it to the working directory, but you shouldn't write it to the datasets directory of galaxy like this. Remember that galaxy is an explicit workflow environment, which means that it controls completely the file naming to abstract it from the workflow implementation process.